### PR TITLE
gui: Keep short deviceID length consistent + xrefs (fixes #9313)

### LINKED ISSUE #9313: Different lengths used for short device IDs in UI
### Does your log mention database corruption?
No

### Include required information

Please be sure to include at least:

 - which version of Syncthing and what operating system you are using
   - 1.27.1, but it's even in `main`
 - what happened
   - IDs are of different length from API and in client in specific sections
 - what you expected to happen instead
   - same-length IDs

https://github.com/syncthing/syncthing/blob/86a08eb87df2aa6cf1ab631120a0b918617a98b3/lib/protocol/deviceid.go#L18 says 7 and therefore from, let's say `/rest/system/connections` I get first 7 characters as if split on `-`
https://github.com/syncthing/syncthing/blob/86a08eb87df2aa6cf1ab631120a0b918617a98b3/gui/default/syncthing/core/syncthingController.js#L1476 and https://github.com/syncthing/syncthing/blob/86a08eb87df2aa6cf1ab631120a0b918617a98b3/gui/default/syncthing/core/syncthingController.js#L1493 however https://github.com/syncthing/syncthing/blob/86a08eb87df2aa6cf1ab631120a0b918617a98b3/gui/default/syncthing/core/syncthingController.js#L1463 has the matching one and https://github.com/syncthing/syncthing/blob/86a08eb87df2aa6cf1ab631120a0b918617a98b3/gui/default/index.html#L913 has even less than both - `5`

Shouldn't it be the same everywhere? Would you welcome a patch or is this something you are aware of and it's kept because of some backward compatibility reason?

### PR DESCRIPTION
**Pull request by KeyWeeUsr**
### Purpose

Making short deviceID length consistent and referencing to protocol file for future-proof edits. Closes #9313.

### CODE REVIEW

File: gui/default/index.html [hunk 1]
```diff
@@ -910,7 +910,7 @@ <h4 class="panel-title">
                       </tr>
                       <tr ng-if="deviceCfg.introducedBy">
                         <th><span class="far fa-fw fa-handshake-o"></span>&nbsp;<span translate>Introduced By</span></th>
-                        <td class="text-right">{{ deviceName(devices[deviceCfg.introducedBy]) || deviceCfg.introducedBy.substring(0, 5) }}</td>
+                        <td class="text-right">{{ deviceName(devices[deviceCfg.introducedBy]) || deviceCfg.introducedBy.substring(0, shortIDStringLength) }}</td>
```
Comment (imsodin) [inline gui/default/index.html:913]: The function `deviceName` already returns a short-id if there is no name, so the or part here does nothing and can be removed. Then you don't need to add `shortIDStringLength` to `$scope`.

Comment (KeyWeeUsr) [reply gui/default/index.html:913]: @imsodin Should I really remove it? There seems to be handling for `undefined` with `""` which is then handled by that `||` and that one isn't present in `deviceName()` :thinking:

Comment (imsodin) [reply gui/default/index.html]: Oh no, you are right - I misread that code. I was thinking the check was done on the same config, but it's `devices[deviceCfg.introducedBy]` vs plain `deviceCfg` afterwards. So lets leave the `||` part, however you can use the existing `deviceShortID` function instead without needing `shortIDStringLength` here.

Comment (KeyWeeUsr) [reply gui/default/index.html]: @imsodin Fixed :v:

File: lib/protocol/deviceid.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ import (
 
 const (
 	DeviceIDLength      = 32
+	// keep consistent with shortIDStringLength in gui/default/syncthing/app.js
 	ShortIDStringLength = 7
```
Comment (imsodin) [inline lib/protocol/deviceid.go:19]: This is the source of the CI failure: The added comment changes the alignment `gofmt -S` expects:

```
=== RUN   TestCheckGoFmt
    gofmt_test.go:39: File ../lib/protocol/deviceid.go is not formatted correctly:
        
        diff ../lib/protocol/deviceid.go.orig ../lib/protocol/deviceid.go
        --- ../lib/protocol/deviceid.go.orig
        +++ ../lib/protocol/deviceid.go
        @@ -14,7 +14,7 @@
         )
         
         const (
        -	DeviceIDLength      = 32
        +	DeviceIDLength = 32
         	// keep consistent with shortIDStringLength in gui/default/syncthing/app.js
         	ShortIDStringLength = 7
         )
--- FAIL: TestCheckGoFmt (1.20s)
```

https://github.com/syncthing/syncthing/actions/runs/7387700143/job/20096964099?pr=9314

Comment (KeyWeeUsr) [reply lib/protocol/deviceid.go]: @imsodin Fixed :+1:

**Pull request merged**